sdk/core/core/src/az_aad.c

```diff
@@ -30,73 +29,20 @@ AZ_NODISCARD az_result _az_token_set(_az_token* self, _az_token const* new_token
   return AZ_OK;
 }
 
-AZ_NODISCARD AZ_INLINE bool _az_url_should_encode(uint8_t c)
-{
-  switch (c)
-  {
-    case '-':
-    case '_':
-    case '.':
-    case '~':
-      return false;
-    default:
-      return !(('0' <= c && c <= '9') || ('A' <= c && c <= 'Z') || ('a' <= c && c <= 'z'));
-  }
-}
-
-// Does not check ref_destination for NULL.
-AZ_NODISCARD az_result _az_url_encode(az_span source, az_span* ref_destination)
-{
-  int32_t const input_size = az_span_size(source);
-
-  int32_t result_size = 0;
-  for (int32_t i = 0; i < input_size; ++i)
-  {
-    result_size += _az_url_should_encode(az_span_ptr(source)[i]) ? 3 : 1;
-  }
-
-  if (az_span_size(*ref_destination) < result_size)
-  {
-    return AZ_ERROR_INSUFFICIENT_SPAN_SIZE;
-  }
-
-  uint8_t* p_s = az_span_ptr(source);
-  uint8_t* p_d = az_span_ptr(*ref_destination);
-  int32_t s = 0;
-  for (int32_t i = 0; i < input_size; ++i)
-  {
-    uint8_t c = p_s[i];
-    if (!_az_url_should_encode(c))
-    {
-      *p_d = c;
-      p_d += 1;
-      s += 1;
-    }
-    else
-    {
-      p_d[0] = '%';
-      p_d[1] = _az_number_to_upper_hex(c >> 4);
-      p_d[2] = _az_number_to_upper_hex(c & 0x0F);
-      p_d += 3;
-      s += 3;
-    }
-  }
-
-  *ref_destination = az_span_slice_to_end(*ref_destination, result_size);
-
-  return AZ_OK;
-}
-
 // https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow#request-an-access-token
 AZ_NODISCARD az_result _az_aad_build_url(az_span url, az_span tenant_id, az_span* out_url)
 {
-  az_span root_url = AZ_SPAN_FROM_STR("https://login.microsoftonline.com/");
+  az_span const root_url = AZ_SPAN_FROM_STR("https://login.microsoftonline.com/");
```

Comment:
This needs to be a configuration setting.  It can change based on which cloud the application is running.  By default we would use the above.
https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-national-cloud

Reply:
Thank you Rick. Yes, I think it must not only be a config setting, but the client secret credential should be configurable for that, i.e. you should be able to create credentilas that use different auth URLs.

Reply:
I updated the issue with my comment above.

Reply:
https://github.com/Azure/azure-sdk-for-c/issues/636